crates/biome_cli/src/cli_options.rs

```diff
@@ -162,3 +168,71 @@ impl Display for CliReporter {
         }
     }
 }
+
+#[derive(Debug, Clone, Copy, Bpaf)]
+pub enum MaxDiagnostics {
+    None,
+    Limit(u32),
```

Comment:
You finally switched back to `u32` instead of `u16`?

Reply:
Yeah. There were useless casts that weren't needed. The only cast we do is from `u32` to `u64`, but that's because is a limit imposed by the LSP library we use.